climada/hazard/tc_tracks.py

```diff
@@ -2980,3 +2980,176 @@ def _zlib_from_dataarray(data_var: xr.DataArray) -> bool:
+            wind_speed = track.max_sustained_wind.values
+            if wind_min and wind_max:
+                index = np.where((wind_speed >= wind_min) & (wind_speed <= wind_max))[0]
+            elif wind_min and not wind_max:
+                index = np.where(wind_speed >= wind_min)[0]
+            elif wind_max and not wind_min:
+                index = np.where(wind_speed <= wind_max)[0]
+            else:
+                index = slice(None)  # select all the track
+
+            # compute 2D density
+            hist_new, _, _ = np.histogram2d(
+                track.lat.values[index],
+                track.lon.values[index],
+                bins=[lat_bins, lon_bins],
+                density=False,
+            )
+            if filter_tracks:
+                hist_new[hist_new > 1] = 1
+
+            hist_count += hist_new
+
+    if norm:
+        hist_count = normalize_hist(res=res, hist_count=hist_count, norm=norm)
+
+    return hist_count, lat_bins, lon_bins
+
+
+def compute_genesis_density(
+    tc_track: TCTracks, lat_bins: np.ndarray, lon_bins: np.ndarray
+) -> np.ndarray:
+    """Compute the density of track genesis locations. This function works under the hood
+    of :py:meth:`compute_track_density`. If it is called with the parameter genesis = True,
+    the function return the number of genesis points per grid cell.
+
+    Parameters:
+    -----------
+
+    tc_track: TCT track object
+        track object containing a list of all tracks
+    lat_bins: 1D np.array
+        array containg the latitude bins
+    lon_bins: 1D np.array
+        array containg the longitude bins
+
+    Returns:
+    --------
+    hist_count: 2D np.array
+        array containing the number of genesis points per grid cell
+    """
+    # Extract the first lat and lon from each dataset
+    first_lats = np.array([ds.lat.values[0] for ds in tc_track.data])
+    first_lons = np.array([ds.lon.values[0] for ds in tc_track.data])
+
+    # compute 2D density of genesis points
+    hist_count, _, _ = np.histogram2d(
+        first_lats,
+        first_lons,
+        bins=[lat_bins, lon_bins],
+        density=False,
+    )
+
+    return hist_count
+
+
+def normalize_hist(
+    res: int,
+    hist_count: np.ndarray,
+    norm: str,
+) -> np.ndarray:
+    """Normalize the number of points per grid cell by the grid cell area or by the total sum of
+    the histogram count.
+
+    Parameters:
+    -----------
+    res: float
+        resolution of grid cells
+    hist_count: 2D np.array
+        Array containing the count of tracks or genesis locations
+    norm: str
+        if norm = area normalize by gird cell area, if norm = sum normalize by the total sum
+    Returns:
+    ---------
+
+    """
+
+    if norm == "area":
+        grid_area, _ = u_coord.compute_grid_cell_area(res=res)
+        norm_hist: np.ndarray = hist_count / grid_area
+    elif norm == "sum":
+        norm_hist: np.ndarray = hist_count / hist_count.sum()
```

Comment:
Perhaps add
```
else:
    raise ValueError("Invalid value for input parameter 'norm', it should be one of 'area' or 'sum'")
```
or something like this? It would also remove the pylint warning below ("Possibly using variable 'norm_hist' before assignment").